sqlmesh/core/context.py

```diff
@@ -517,41 +526,53 @@ def state_reader(self) -> StateReader:
 
     def refresh(self) -> None:
         """Refresh all models that have been updated."""
-        if self._loader.reload_needed():
+        if any(loader.reload_needed() for loader, _ in self._loaders.values()):
             self.load()
 
     def load(self, update_schemas: bool = True) -> GenericContext[C]:
         """Load all files in the context's path."""
         load_start_ts = time.perf_counter()
-        with sys_path(*self.configs):
-            project = self._loader.load(self, update_schemas)
-            self._macros = project.macros
-            self._jinja_macros = project.jinja_macros
-            self._models = project.models
-            self._metrics = project.metrics
-            self._standalone_audits.clear()
-            self._audits.clear()
+
+        projects = []
+        for loader, configs in self._loaders.values():
+            with sys_path(*configs):
+                projects.append(loader.load(self, update_schemas))
+
+        self._standalone_audits.clear()
+        self._audits.clear()
+        self._macros.clear()
+        self._models.clear()
+        self._metrics.clear()
+        for project in projects:
+            self._jinja_macros = self._jinja_macros.merge(project.jinja_macros)
+            self._macros.update(project.macros)
+            self._models.update(project.models)
+            self._metrics.update(project.metrics)
+
             for name, audit in project.audits.items():
                 if isinstance(audit, StandaloneAudit):
                     self._standalone_audits[name] = audit
                 else:
                     self._audits[name] = audit
-            self.dag = project.dag
 
-            duplicates = set(self._models) & set(self._standalone_audits)
-            if duplicates:
-                raise ConfigError(
-                    f"Models and Standalone audits cannot have the same name: {duplicates}"
-                )
+        self.dag = (
+            DAG({**projects[0].dag.graph, **projects[1].dag.graph})
+            if len(projects) == 2
+            else projects[0].dag
+        )
```

Comment:
Should we just do a loop over `projects` here and generalize this logic instead of indexing `[0]` and `[1]`? So, `self.dag = <expand all `dag.graph` project attributes>` or something.

Reply:
right missed this thanks!